recipes/libcublas/meta.yaml

```diff
@@ -0,0 +1,96 @@
+{% set name = "libcublas" %}
+{% set version = "12.0.1.189" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 38272cc9184b4a57a6250fc2cd914ffc827f9f8cd5a3e1356bc3c31f843caf8f  # [linux64]
+  sha256: 2d46e735f8e76e2eb7df1b4f41d56aa8d5d2683f99988bac82d6ae33fb6d344b  # [ppc64le]
+  sha256: 96b68f6821684f39f338862639ce74a74c75830bf44dcd17ef372a1cb8c2304d  # [aarch64]
+  sha256: 88f29bec81880844da2eee13f1cdc51e1ccd8b4c60af6df633577810b30f80d0  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:
+    strong:
+      - __glibc >=2.17
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+
+outputs:
+  - name: libcublas
+    files:
+      - lib/lib*blas*.so.*            # [linux]
+      - targets/*/lib/lib*blas*.so.*  # [linux]
+      - Library\bin                   # [win]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcublas.so.*  # [unix]
+        - if not exist %LIBRARY_LIB%\\cublas.lib exit 1  # [win]
+
+  - name: libcublas-dev
+    requirements:
+      run_exports:
+        - {{ pin_subpackage("libcublas", max_pin="x") }}
+    files:
+      - lib/lib*blas*.so            # [linux]
+      - targets/*/lib/lib*blas*.so  # [linux]
+      - lib/stubs                   # [linux]
+      - include                     # [linux]
+      - targets/*/include/*         # [linux]
+      - lib/pkgconfig               # [linux]
+      - src
+      - Library\include             # [win]
+      - Library\lib                 # [win]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcublas.so  # [unix]
+        - test -f $PREFIX/include/cublas.h  # [unix]
+        - if not exist %LIBRARY_LIB%\\cublas.lib exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\cublas.h exit 1  # [win]
+
+  - name: libcublas-static
+    build:
+      skip: True  # [not linux]
+    files:
+      - lib/lib*blas*_static.a
+      - targets/*/lib/lib*blas*_static.a
```

Comment:
Could you explain a bit why it should depend on the `dev` (or any) package? I assumed that the use case for the `static` package is people who want to use it at build time to statically link against `libcublas` so that they don't have any runtime dependence. 

Do you need more than just the `.a` to link to it?

Reply:
I would think linking a static library would still need the headers from the `dev` package to be useful. If that's not the case we could drop the dependency.

Reply:
Yeah think this goes back to how we want to define `-dev` & `-static` that we have had in a couple other places ( https://github.com/conda-forge/staged-recipes/pull/21902#discussion_r1092460458 ) ( https://github.com/conda-forge/staged-recipes/pull/21723#discussion_r1084611245 )

Thus far we've gone with `-dev` should pull in everything except `-static`, which is constrained, but not required. Though an exception to exception is `cuda-cudart`, in which `-dev` pulls in the `-static` portion too

Reply:
I guess to compile code that relies on things in the static library, you need the headers. But in that case you should be asking for `-dev` no?

In https://github.com/conda-forge/staged-recipes/pull/21723#discussion_r1084611245 the conclusion was that `-dev` should require `-static`. `-static` doesn't require anything else.

The way to use this for a package that wants to link to the static library then is to require the `-dev` package at build time, link to the static library, produce the build output and be done. The finished executable would not have any run time dependency on `-dev` or `-static`.

So I'd add `-static` as a dependency of `-dev`, but leave `-static` as is (unlike the suggestion which suggests adding `-dev` as a requirement for `-static`).

Reply:
Yeah I think you have the right of it. Lets follow https://github.com/conda-forge/staged-recipes/pull/21902#discussion_r1092460458 for adding the `run_constrained` in the `-dev` package and leaving `-static` with no dependency.

Reply:
Done. I'll mark this thread as resolved, so that they don't get super long. We can start a new one if there is a new thing to discuss.